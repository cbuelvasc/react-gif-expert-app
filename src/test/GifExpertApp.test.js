import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test GifExpertApp component', () => {

    let wrapper = shallow(<GifExpertApp />);

    beforeEach(() => {
        wrapper = shallow(<GifExpertApp />);
    });

    test('Component GifExpertApp should show correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Component GifExpertApp should show a list of categories', () => {
        const categories = ['One punch', 'Dragon Ball'];
        wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});