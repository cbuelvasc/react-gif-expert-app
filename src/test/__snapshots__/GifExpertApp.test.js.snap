// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpertApp component Component GifExpertApp should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Test GifExpertApp component Component GifExpertApp should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
